pkg/acsengine/transform/transform.go

```diff
@@ -155,6 +155,11 @@ func (t *Transformer) NormalizeForK8sVMASScalingUp(logger *logrus.Entry, templat
 	if err := t.NormalizeMasterResourcesForScaling(logger, templateMap); err != nil {
 		return err
 	}
+	return RemoveNSGs(t, logger, templateMap)
+}
+
+// RemoveNSGs removes the NSG from the templateMap
+func RemoveNSGs(t *Transformer, logger *logrus.Entry, templateMap map[string]interface{}) error {
```

Comment:
This new function doesn't just remove NSG resources, it also removes route tables and vnets. Is that the intention?

Reply:
@jackfrancis - Indeed this is just a WIP. Still need to extract vnet and route tables and ensure that NSG removal is done only once.